e2e-pw/src/oss/poms/saved-views.ts

```diff
@@ -0,0 +1,406 @@
+import { expect, Locator, Page } from "src/oss/fixtures";
+
+const ColorList = [
+  "Gray",
+  "Blue",
+  "Purple",
+  "Red",
+  "Yellow",
+  "Green",
+  "Pink",
+  "Orange",
+  "Purple",
+];
+type Color =
+  | "Gray"
+  | "Blue"
+  | "Purple"
+  | "Red"
+  | "Yellow"
+  | "Green"
+  | "Pink"
+  | "Orange"
+  | "Purple";
+
+const updatedView = {
+  name: "test updated",
+  description: "test updated",
+  color: "Yellow" as Color,
+};
+
+export const updatedView2 = {
+  name: "test updated 2",
+  description: "test updated 2",
+  color: "Orange" as Color,
+};
+
+export class SavedViewsPom {
+  readonly page: Page;
+  readonly assert: SavedViewAsserter;
+
+  readonly locator: Locator;
+  readonly dialogLocator: Locator;
+
+  savedViewNameA = "test name 1";
+
+  constructor(page: Page) {
+    this.page = page;
+    this.assert = new SavedViewAsserter(this);
+
+    this.locator = page.getByTestId("saved-views-select-container");
+    this.dialogLocator = page.getByTestId("saved-views-modal-container");
+  }
+
+  clearSavedViewsStmt() {
+    return `
+      import fiftyone as fo
+      dataset = fo.load_dataset("${this.savedViewNameA}")
+      dataset.delete_saved_views()
+    `;
+  }
+
+  async clickEditRaw(name: string = "test") {
+    await this.locator.click();
+    await this.clickOptionEdit(name);
+  }
+
+  async clickOptionEdit(name: string = "test") {
+    await this.savedViewOption(name).hover();
+    await this.optionEdit(name).click();
+  }
+
+  async clickEdit(name: string = "test") {
+    await this.clearView();
+    await this.clickEditRaw(name);
+  }
+
+  optionEdit(name: string = "test") {
+    return this.savedViewOption(name).getByTestId("btn-edit-selection");
+  }
+
+  async saveViewInputs(
+    name: string = "test",
+    description: string = "test",
+    color: Color = "Gray",
+    newColor: Color = "Blue"
+  ) {
+    await this.nameInput().type(name);
+    await this.page.waitForTimeout(500);
+    await this.descriptionInput().type(description);
+    await this.colorInput(color).click();
+    await this.colorOption(newColor).click();
+  }
+
+  async saveView(
+    name: string = "test",
+    description: string = "test",
+    color: Color = "Gray",
+    newColor: Color = "Blue"
```

Comment:
I'd argue that these default values (other than color maybe) shouldn't be part of a POM